mars_troughs/accumulation_model.py

```diff
@@ -31,41 +28,203 @@ class InsolationAccumulationModel(AccumulationModel):
+        self._cot = np.cos(self._angle) * self._csc
 
+    @property
+    def csc_angle(self) -> float:
+        """
+        Cosecant of the slope angle.
+        """
+        return self._csc
 
-class LinearInsolationAccumulation(InsolationAccumulationModel):
+    @property
+    def cot_angle(self) -> float:
+        """
+        Cotangent of the slope angle.
+        """
+        return self._cot
+
+class LinearAccuIns(AccumulationDepInsolation):
     """
     Accumulation is linear in solar insolation.
 
     Args:
         times (np.ndarray): times at which the solar insolation is known
         insolations (np.ndarray): value of the solar insolations
-        intercept (float, optional): default is 0 millimeters
-        slope (float, optional): default is 1e-6 mm per unit
-            of solar insolation square.
+        intercept (float, optional): default is 0 m
+        slope1 (float, optional): default is 1e-6 m per unit
+            of solar insolation.
     """
+    def __init__(self, times, insolation, intercept: float = 0.0, slope: float = 1e-6):
+        
+        super().__init__(times, insolation)
+        
+        self.intercept=intercept
+        self.slope=slope
+        
+    def get_accumulation_at_t(self, time: np.ndarray):
+        """
+        Accumulation rate A (in m/year) as a  linear function of insolation: 
+        A(ins(t)) = intercept + slope*ins(t).  Intercept in m/year and slope in 
+        m^3/(year*W).
 
-    def __init__(
-        self,
-        times: np.ndarray,
-        insolations: np.ndarray,
-        intercept: float = 1.0,
-        slope: float = 1e-6,
-    ):
-        super().__init__(times, insolations)
-        self.intercept = intercept
-        self.slope = slope
+        Args:
+            time (np.ndarray): times at which we want to calculate A, in years.
+        Output:
+            np.ndarray of the same size as time input containing values of 
+            accumulation rates A in m/year
+        
+        """
+        return self.intercept -1 * (self.slope * self.ins_data_spline(time))
+    
+    def get_yt(self, time: np.ndarray):
+        """
+        Calculates the vertical distance y (in m) at times t traveled by a point
+        in the center of the high side of the trough. This distance  is a 
+        function of the accumulation rate A as y(t)=int(A(ins(t)) dt) or 
+        dy/dt=A(ins(t))
+
+        Args:
+            time (np.ndarray): times at which we want to calculate y, in years.
+        Output:
+            np.ndarray of the same size as time input containing values of 
+            the vertical distance y, in meters.
+        
+        """
+        return self.intercept -1 * (self.slope * (self.iins_data_spline(time) 
+                                                 - self.iins_data_spline(0)))
+    
+    
+        
+class SquareAccuIns(AccumulationDepInsolation):
+    """
+    Accumulation is linear in solar insolation.
+
+    Args:
```

Comment:
Missing a docstring for `slope2`. Also, the docstring for `slope1` should say it's for the linear term, while the docstring for `slope2` should say it's for the quadratic term.